sklearn/callback/_progressbar.py

```diff
@@ -0,0 +1,187 @@
+# License: BSD 3 clause
+# Authors: the scikit-learn developers
+
+import importlib
+from multiprocessing import Manager
+from threading import Thread
+
+from . import BaseCallback
+
+
+class ProgressBar(BaseCallback):
+    """Callback that displays progress bars for each iterative steps of an estimator"""
+
+    auto_propagate = True
+
+    def __init__(self):
+        try:
+            importlib.import_module("rich")  # noqa
+        except ImportError as e:
+            raise ImportError("ProgressBar requires rich installed.") from e
+
+    def on_fit_begin(self, estimator, X=None, y=None):
+        self._queue = Manager().Queue()
+        self.progress_monitor = _RichProgressMonitor(queue=self._queue)
+        self.progress_monitor.start()
+
+    def on_fit_iter_end(self, *, estimator, node, **kwargs):
+        self._queue.put(node)
+
+    def on_fit_end(self):
+        self._queue.put(None)
+        self.progress_monitor.join()
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        if "progress_monitor" in state:
```

Comment:
What is the reason for deleting this part from the state here. Is it when unpickling an estimator to have a clean estimator? It could be worth to have a comment to explain.